R/geom-hex.r

```diff
@@ -56,12 +56,33 @@ geom_hex <- function(mapping = NULL, data = NULL,
 GeomHex <- ggproto("GeomHex", Geom,
   draw_group = function(data, panel_params, coord, lineend = "butt",
                         linejoin = "mitre", linemitre = 10) {
-    if (!inherits(coord, "CoordCartesian")) {
-      abort("geom_hex() only works with Cartesian coordinates")
+    if (empty(data)) {
+      return(zeroGrob())
     }
 
+    # Get hex sizes
+    if (!is.null(data$width)) {
+      dx <- data$width[1] / 2
+    } else {
+      dx <- resolution(data$x, FALSE)
+    }
+    if (!is.null(data$height)) {
+      dy <- data$height[1] /  sqrt(3) / 2
+    } else {
+      dy <- resolution(data$y, FALSE) / sqrt(3) / 2 * 1.15
```

Comment:
Do we have an explanation for the magic numbers here (sqrt(3), 2, 1.15)? I realize they're just copied over from the old code, but some comment would be great.

Also, the factor 1.15 is missing in line 70, and this is different from the old code. Could you add a comment explaining the difference between line 70 and line 72?

Otherwise looks good.

Reply:
I think it is simply a matter of calculating the relation between the height and width of a regular hexagon. The 1.15 is there because resolution in the y-direction is a different concept to the binwidth (bins are shifted so they overlap in the y-axis, affecting the resolution)... I can add some comment on this of course